### ISSUE TITLE
Layer geoserver wfs:imos:srs_oc_npp_vgpm_epp_gsm_1d_aqua_url is slow (and all other srs_oc_* layers)

### ISSUE THREAD

**Issue by danfruehauf** [aodn/content#86]
Layer starts only after 3-4 seconds normally:
https://nagios.aodn.org.au/pnp4nagios/graph?host=5-nec-mel.emii.org.au&srv=geoserver+wfs%3Aimos%3Asrs_oc_npp_vgpm_epp_gsm_1d_aqua_url&view=2&start=&end=

**Comment by danfruehauf** [comment]
Changed title to suggest all SRS_OC layers are slow.

**Comment by danfruehauf** [comment]
The slowness of those layers is a result of heavily relying on regular expressions when generating the views. For example:
```
SELECT t.id AS timeseries_id,
    f.url AS file_url,
    f.size,
    timezone('UTC'::text, t."time") AS "time"
   FROM srs_oc.timeseries t
     JOIN srs_oc.indexed_file f ON t.file_id = f.id
  WHERE f.url::text ~ '.*/A[0-9]{14}.L3m_MO_SO_Chl_9km.Johnson_SO_Chl.nc4'::text
  ORDER BY timezone('UTC'::text, t."time");
```

I suggest relying on something a bit more rudimentary which can be indexed much better, such as:
```
  SELECT t.id AS timeseries_id,
    f.url AS file_url,
    f.size,
    timezone('UTC'::text, t."time") AS "time"
   FROM srs_oc.timeseries t
     JOIN srs_oc.indexed_file f ON t.file_id = f.id
  WHERE f.url like '%/A%.L3m_MO_SO_Chl_9km.Johnson_SO_Chl.nc4'
  ORDER BY timezone('UTC'::text, t."time");
```

It this is not enough, then perhaps adding another key in the `indexed_file` table to indicate the view the file belongs to will be another option.

**Comment by danfruehauf** [comment]
Another thing which can be nice to have is an index on `timeseries.file_id`, such as:
```
CREATE INDEX timeseries_file_id_idx
  ON srs_oc.timeseries
  USING btree
  (file_id);
```

**Comment by lbesnard** [comment]
https://github.com/aodn/harvesters/pull/226

**Comment by danfruehauf** [comment]
Unfortunately still under-performing in RC.

I think we'll have to go with the enum solution as it can be indexed much better when presenting the views.

**Comment by danfruehauf** [comment]
I take it back. It is indeed much faster now and also deployed to prod. So Fixed as well.
